File: desktop/sdk-macos/Sources/EatPassDesktop/EatPassConfig.swift
import Foundation

public struct EatPassConfig: Sendable {
    public let attesterUrl: String
    public let issuerUrl: String
    public let issuerName: String
    public let originInfo: String
    public let ktLogPubHex: String?
    public let timeoutSeconds: TimeInterval
    public let teamId: String
    public let bundleId: String

    public init(
        attesterUrl: String,
        issuerUrl: String,
        issuerName: String = "issuer.eat-pass.dev",
        originInfo: String = "tool-gate.secure.build/v1/tools/email.send",
        ktLogPubHex: String? = nil,
        timeoutSeconds: TimeInterval = 30,
        teamId: String,
        bundleId: String
    ) {
        self.attesterUrl = attesterUrl
        self.issuerUrl = issuerUrl
        self.issuerName = issuerName
        self.originInfo = originInfo
        self.ktLogPubHex = ktLogPubHex
        self.timeoutSeconds = timeoutSeconds
        self.teamId = teamId
        self.bundleId = bundleId
    }
}

public struct MintResult: Sendable {
    public let authorizationHeader: String
    public let bindingHex: String
}

public enum EatPassError: Error, LocalizedError {
    case http(String)
    case attestation(String)
    case crypto(String)

    public var errorDescription: String? {
        switch self {
        case .http(let m), .attestation(let m), .crypto(let m): return m
        }
    }
}

File: mobile/sdk-android/eatpass-mobile/src/main/java/com/securebuild/eatpass/EatPassConfig.kt
package com.securebuild.eatpass

/**
 * Endpoints and identifiers for the coupled mobile gate (attestation + eat-pass mint).
 */
data class EatPassConfig(
    val attesterUrl: String,
    val issuerUrl: String,
    val issuerName: String = "issuer.eat-pass.dev",
    val originInfo: String = "tool-gate.secure.build/v1/tools/email.send",
    /** Optional: reject issuer if KT log pubkey does not match. */
    val ktLogPubHex: String? = null,
    /** HTTP connect/read timeout in seconds. */
    val timeoutSeconds: Long = 30,
)

data class MintResult(
    /** Ready-to-send `Authorization` header value (PrivateToken). */
    val authorizationHeader: String,
    val bindingHex: String,
)

class EatPassException(message: String, cause: Throwable? = null) : Exception(message, cause)
